caffe2/proto/torch.proto

```diff
@@ -4,6 +4,46 @@ import "caffe2/proto/caffe2.proto";
 
 package torch;
 
+message TensorDef {
+  // the tensor id in ModelDef's tensor table
+  optional string tensor_id = 1;
+  repeated int64 dims = 2;
+  optional int64 offset = 3;
+  repeated int64 strides = 4;
+
+  // the same as caffe2.TensorProto.DataType
+  // 0: UNDEFINED
+  // 1: FLOAT
+  // 2: INT32
+  // 3: BYTE
+  // 4: STRING
+  // 5: BOOL
+  // 6: UINT8
+  // 7: INT8
+  // 8: UINT16
+  // 9: INT16
+  // 10: INT64
+  // 12: FLOAT16
+  // 13: DOUBLE
+  optional int32 data_type = 5;
+
+  // the same as caffe2.ExternalDataProto.SourceType
```

Comment:
you can kill it from caffe2.proto then I guess (as we won't be keeping the extern data there explicitly)

also, we could just move the enum here